diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter2/kdoc/KdocMethods.kt

```diff
@@ -146,6 +148,14 @@ class KdocMethods(configRules: List<RulesConfig>) : DiktatRule(
         }
     }
 
+    private fun isReferenceExpressionWithSameName(node: ASTNode, kdocTags: Collection<KDocTag>?): Boolean {
+        if (node.isSingleLineGetterOrSetter()) {
+            return false
+        }
+        val hasReturnKdoc = kdocTags != null && kdocTags.hasKnownKdocTag(KDocKnownTag.RETURN)
+        return node.findAllDescendantsWithSpecificType(REFERENCE_EXPRESSION).map { it.text }.lastOrNull() == (node.psi as KtFunction).name // || !hasReturnKdoc
```

Comment:
I suggest to check not any descendant, but precisely the last reference in dot qualified expression. For more complex cases we still would like to add KDoc, e.g.
```kotlin
fun getProjects() = service.getProjects().map { transofrm(it) }
```
this is not very self-explanatory example

Reply:
Ok. Fixed it